datastore/client.go

```diff
@@ -90,13 +106,27 @@ func (dc *datastoreClient) Rollback(ctx context.Context, in *pb.RollbackRequest,
 }
 
 func (dc *datastoreClient) AllocateIds(ctx context.Context, in *pb.AllocateIdsRequest, opts ...grpc.CallOption) (res *pb.AllocateIdsResponse, err error) {
+	ctx = trace.StartSpan(ctx, "cloud.google.com/go/datastore.AllocateIds")
+	defer func() { trace.EndSpan(ctx, err) }()
+
 	err = dc.invoke(ctx, func(ctx context.Context) error {
 		res, err = dc.c.AllocateIds(ctx, in, opts...)
 		return err
 	})
 	return res, err
 }
 
+func (dc *datastoreClient) ReserveIds(ctx context.Context, in *pb.ReserveIdsRequest, opts ...grpc.CallOption) (res *pb.ReserveIdsResponse, err error) {
```

Comment:
I would leave this out for now; it can go in a separate change if the datastore team wants it to be added.